src/GraphQL/Types/QueryArgument.cs

```diff
@@ -117,6 +117,32 @@ public string Name
             return type;
         }
 
+        /// <summary>
+        /// Parses the value received from the client when the value is not <see langword="null"/>.
+        /// Occurs during validation prior to <see cref="Validator"/>.
+        /// Throw an exception if necessary to indicate a problem.
+        /// Only applicable to fields of input graph types.
```

Comment:
Is there a plan to finish the #3583 for v8? If not, it may be worth creating an analyzer to error on incorrect usage. Or at least throw a runtime exception during initialization.

Reply:
> Is there a plan to finish the https://github.com/graphql-dotnet/graphql-dotnet/pull/3583 for v8?

I don't think so.  I don't have the time to work on it, and it would probably cause a number of breaking changes anyway, depending on the final implementation.

> If not, it may be worth creating an analyzer to error on incorrect usage. Or at least throw a runtime exception during initialization.

I agree; maybe even both.  So throw an exception during initialization if `fieldType.Parser` or `fieldType.Validator` was set for an input type?

FYI the advantage of not throwing an exception during initialization (and only using an analyzer) is so that dynamic code (such as the auto-registering types' code) need not know if it's an input field or not.  So far within GraphQL.NET this isn't a problem.

I'll add the initialization check and you can comment if you think it's better without it.  If you have time you can write an analyzer, or we can just stick with the initialization check.

Reply:
We already have checks to be sure Resolver and StreamResolver are not set for input fields, so certainly we should add these checks.

Reply:
I think the exception should be there because no one loves silent errors. This is a new feature, so when people start using it, they will immediately find they used it wrongly and will read the documentation more carefully. Adding an exception in a later stage is usually a problem because the code that is used to work will start throwing after version upgrading. But the analyzer can be added even after the v8.0 release because it will cause compile time errors only on newly added parsers and validators but not on existing ones.

Reply:
And you can always silence an analyzer warning/error.

Reply:
Schema initialization checks have been added.